mantidproject/mantid
# Engg GUI::Enable Fitting tab to Select/Add/Save peaks & zoom-in/out option

**Description of work.**

This branch improves the `Preview` section from the `Fitting tab`, upon completion of Fitting process, an option to `select`, `add` and `save` peak from the data plot of the focused workspace will be enabled. The peaks selected will be added to the `Peak` text-field so it can be utilised to either rerun `Fit` or save the peaks as _CSV_ file. The branch also supports option to `zoom-in` or `zoom-out` of the data plot once plots have been fitted, more details description can be found in documentation file.

**To test:**
Files can be found at: `\\olympic\Babylon5\Public\Shahroz\`
Select a focused file to Fit : `ENGINX_241391_focused_texture_bank_3.nxs`
Add peaks: `3.1243,2.7057,1.9132` _or leave blank_
Check with empty plot:
 `Select Peak` and `Add Peak` buttons are disabled
 `Zoom-in` feature is disabled
Click `Fit`

Once Fit process completed:
1. Test the `zoom-in` feature
   `zoom-in` by left click and drag on plot
   `zoom-out` by right click on plot
2. Click `Select Peak` button
   This should enable the `peak picking tool`
   Test feature by holding `shift` key and `left click` on graph
   Increase the width of the tool by using `Ctrl` key and dragging horizontally
3. Once centre of peak selected:
   Click `Add Peak` button
   Which should add the selected peak to `Peaks` text-field 
   The integer should be to 4 d.p.
   _(Units are required to be changed to dSpacing)_ #15766
   Unless it can be `Fit` again to plot
   or
4. Peaks can be saved as `.csv` file by clicking `Save Peak List` button

Documentation update
Code Check



Fixes #15765.



[Release Notes](https://github.com/mantidproject/mantid/blob/15ab5125dc809b610c34fb059b2cd449178fc6f0/docs/source/release/v3.7.0/diffraction.rst)

---
#### Reviewer

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):
##### Code Review
##### Functional Tests
- Does everything look good? Comment with the ship it emoji but don't merge. A member of `@mantidproject/gatekeepers` will take care of it.
